var/spack/repos/builtin/packages/zfp/package.py

```diff
@@ -62,55 +73,80 @@ class Zfp(CMakePackage, CudaPackage):
+    variant("profile", default=False, when="@0.5.2:", description="Count cache misses")
 
-    # Dependencies
-    depends_on("cmake@3.4.0:", type="build")
-    depends_on("cuda@7:", type=("build", "test", "run"), when="+cuda")
-    depends_on("python", type=("build", "test", "run"), when="+python")
-    depends_on("py-numpy", type=("build", "test", "run"), when="+python")
-    depends_on("py-cython", type="build", when="+python")
+    variant(
+        "round",
+        default="never",
+        values=("never", "first", "last"),
+        multi=False,
+        when="@1.0.0:",
+        description="EXPERIMENTAL: Set coefficient rounding method",
+    )
+
+    variant(
+        "strided", default=False, description="Enable strided access for progressive zfp streams"
+    )
+
+    variant(
+        "tight-error",
+        default=False,
+        when="@1.0.0:",
+        description="EXPERIMENTAL: Use tighter error bound when rounding is enabled",
+    )
+
+    variant("twoway", default=False, description="Use two-way skew-associative cache")
+
+    # Conflicts
+    conflicts(
+        "+tight-error",
+        when="round=never",
+        msg="Using zfp with tight error requires a rounding mode other than never",
+    )
 
+    # CMake options
     def cmake_args(self):
         spec = self.spec
 
-        # CMake options
         args = [
-            # TODO: self.define_from_variant('BUILD_UTILITIES', 'utilities'),
             self.define("BUILD_TESTING", self.run_tests),
             self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant("BUILD_UTILITIES", "utilities"),
             self.define_from_variant("BUILD_CFP", "c"),
             self.define_from_variant("BUILD_ZFPY", "python"),
             self.define_from_variant("BUILD_ZFORP", "fortran"),
             self.define_from_variant("ZFP_WITH_OPENMP", "openmp"),
             self.define_from_variant("ZFP_WITH_CUDA", "cuda"),
-            self.define("ZFP_BIT_STREAM_WORD_SIZE", spec.variants["bsws"].value),
-            self.define_from_variant("ZFP_WITH_BIT_STREAM_STRIDED", "strided"),
             self.define_from_variant("ZFP_WITH_ALIGNED_ALLOC", "aligned"),
-            self.define_from_variant("ZFP_WITH_CACHE_TWOWAY", "twoway"),
+            self.define("ZFP_BIT_STREAM_WORD_SIZE", spec.variants["bsws"].value),
+            self.define_from_variant("ZFP_WITH_DAZ", "daz"),
             self.define_from_variant("ZFP_WITH_CACHE_FAST_HASH", "fasthash"),
             self.define_from_variant("ZFP_WITH_CACHE_PROFILE", "profile"),
+            self.define_from_variant("ZFP_WITH_BIT_STREAM_STRIDED", "strided"),
+            self.define_from_variant("ZFP_WITH_TIGHT_ERROR", "tight-error"),
+            self.define_from_variant("ZFP_WITH_CACHE_TWOWAY", "twoway"),
         ]
 
+        if "round" in spec.variants:
+            args.append("ZFP_ROUNDING_MODE={0}".format(spec.variants["round"].value.upper()))
```

Comment:
I may be missing something obvious here, but don't we need to prepend `ZFP_ROUND_` to the value after conversion to upper case?  For example, Spack's `round=never` should translate to CMake's `ZFP_ROUNDING_MODE=ZFP_ROUND_NEVER`.